File: Submissions/1015_Smallest_Integer_Divisible_by_K.cpp
class Solution {
public:
    int smallestRepunitDivByK(int K) {
        if(K%2 == 0 || K%5 == 0) return -1;
        int N = 1;
        for(int i = 1; ;++i){
            if(N%K == 0) return i;
            N = (N * 10 + 1) % K;
        }
        
        return -1;
    }
};

File: Kotlin/1015_Smallest_Integer_Divisible_by_K.kt
class Solution {
    fun smallestRepunitDivByK(K: Int): Int {
        if (K % 2 == 0 || K % 5 == 0) return -1
        var result = 1
        var n = 1
        while(n % K != 0){
            n = (n * 10 + 1) % K
            ++result
        }
        
        return result
    }
}